src/pharmpy/statements.py

```diff
@@ -593,6 +590,38 @@ def set_lag_time(self, compartment, lag_time):
         return new_comp
 
 
+def _is_positive(expr: sympy.Expr) -> bool:
+    return sympy.ask(
+        sympy.Q.positive(expr), _assume_all(sympy.Q.positive, _free_images_and_symbols(expr))
+    )
+
+
+def _free_images_and_symbols(expr: sympy.Expr) -> Set[sympy.Expr]:
+    return expr.free_symbols | _free_images(expr)
+
+
+def _free_images(expr: sympy.Expr) -> Set[sympy.Expr]:
+    return set(_free_images_iter(expr))
+
+
+def _free_images_iter(expr: sympy.Expr) -> Iterable[sympy.Expr]:
+    if isinstance(expr.func, sympy.core.function.UndefinedFunction):
```

Comment:
Is this needed because we can have symbolic functions in expressions?

Reply:
Yes! It targets `A_i(t)` "images".

Reply:
Ok